src/TLuaInterpreter.cpp

```diff
@@ -6754,21 +6754,24 @@ int TLuaInterpreter::sendMSDP(lua_State* L)
 {
     Host& host = getHostFromLua(L);
     std::string variable;
-    if (!lua_isstring(L, 1)) {
-        lua_pushfstring(L, "sendMSDP: bad argument #1 type (variable name as string expected, got %1!)", luaL_typename(L,1));
-        return lua_error(L);
-    } else {
-        variable = host.mTelnet.encodeAndCookBytes(lua_tostring(L, 1));
+    int n = lua_gettop(L);
+    
+    for (int i = 1; i <= n; ++i) {
+        if (!lua_isstring(L, i)) {
+            lua_pushfstring(L, "sendMSDP: bad argument type #" + std::to_string(i) + " type (variable name as string expected, not %1!)", luaL_typename(L, 1));
```

Comment:
Four things:
* The original `lua_pushfstring(...)` format string is in error - and you have copied it - it uses a `%s` for a `const char` type string replacement - NOT a `%1` i.e. with a number - *that* is for `QString::arg(...)`s... :man_facepalming: 
* You need to refer to the correct argument when you are reporting the type with `luaL_typename(...)` so you need a variable (`i` in this case) not a `1`! :stuck_out_tongue_winking_eye:
* Only the first argument is a "variable", all the rest are "value"s :open_mouth: 
* There is a simple means to introduce a string or number into a `lua_pushfstring(...)` - by using replacement parameters (like C `printf`) :nerd_face: 

So I'd suggest the following for this line instead:
```cpp
lua_pushfstring(L, "sendMSDP: bad argument type #%d type (%s as string expected, got %s!)", i, (i == 1 ? "variable name" : "value"), luaL_typename(L, i));
```
HTH...